src/test/java/gov/usgs/wma/waterdata/groundwater/BuildRdbFileTest.java

```diff
@@ -215,6 +217,64 @@ protected RdbWriter createRdbWriter(Writer destination) {
 		assertTrue(dstWriterClosed);
 	}
 
+	@Test
+	void testEmptyRdbException() throws Exception {
+		// SETUP
+
+		S3Bucket mockS3b = Mockito.mock(S3Bucket.class);
+		mockS3b.writer = destination;
+		Mockito.when(mockS3b.getWriter()).thenReturn(destination);
+		Mockito.doCallRealMethod().when(mockS3b).close();
+
+		S3BucketUtil mockS3u = Mockito.mock(S3BucketUtil.class);
+		Mockito.when(mockS3u.createFilename(POSTCD)).thenReturn(FILENM);
+		Mockito.when(mockS3u.openS3(FILENM)).thenReturn(mockS3b);
+
+		DiscreteGroundWaterDao mockDao = Mockito.mock(DiscreteGroundWaterDao.class);
+		AqToNwisParmDao mockAqDao = Mockito.mock(AqToNwisParmDao.class);
+
+		LocationFolder mockLoc = Mockito.mock(LocationFolder.class);
+		Mockito.when(mockLoc.getLocationFolders()).thenReturn(stateAsList);
+		Mockito.when(mockLoc.toStates(STATE)).thenReturn(stateAsList);
+		Mockito.when(mockLoc.filenameDecorator(STATE)).thenReturn(POSTCD);
+		Mockito.when(mockAqDao.getParameters()).thenReturn(getParameterList());
+
+		BuildRdbFile builder = new BuildRdbFile() {
+			@Override
+			protected RdbWriter createRdbWriter(Writer destination) {
+				writer = Mockito.mock(RdbWriter.class);
+				Mockito.when(writer.writeHeader()).thenReturn(writer);
+				Mockito.when(writer.getDataRowCount()).thenReturn(0L);
+				return writer;
+			}
+		};
+		builder.aqDao = mockAqDao;
+		builder.dao = mockDao;
+		builder.s3BucketUtil = mockS3u;
+		builder.locationFolderUtil = mockLoc;
+
+		// ACTION UNDER TEST
+		RuntimeException e = assertThrows(RuntimeException.class, ()->builder.apply(req), "Expected RuntimeException to have been thrown");
+		assertEquals(EMPTY_RDB_MSG, e.getMessage());
+
+		// ASSERTIONS
+		Mockito.verify(mockAqDao, Mockito.atLeastOnce()).getParameters();
+		Mockito.verify(mockAqDao, Mockito.atMostOnce()).getParameters();
```

Comment:
Is there a method on `Mockito` like `once()` or something similar? It seems like we're testing that something about these parameters happened exactly 1 time, but I could be totally wrong.

Reply:
I think there is a times(1) that would achieve that.